tools/prom-metric-linter/metrics_collector.go

```diff
@@ -0,0 +1,136 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2023 Red Hat, Inc.
+ *
+ */
+
+package main
+
+import (
+	"fmt"
+	"os"
+	"regexp"
+	"strings"
+
+	dto "github.com/prometheus/client_model/go"
+)
+
+// Metric represents a Prometheus metric
+type Metric struct {
+	Name string
+	Help string
+	Type string
+}
+
+// excludedMetrics defines the metrics to ignore, open issue:https://github.com/kubevirt/kubevirt/issues/9714
+// Do not add metrics to this list!
+var excludedMetrics = map[string]string{
+	"kubevirt_allocatable_nodes_count":                     "",
+	"kubevirt_kvm_available_nodes_count":                   "",
+	"kubevirt_migrate_vmi_pending_count":                   "",
+	"kubevirt_migrate_vmi_running_count":                   "",
+	"kubevirt_migrate_vmi_scheduling_count":                "",
+	"kubevirt_virt_api_up_total":                           "",
+	"kubevirt_virt_controller_ready_total":                 "",
+	"kubevirt_virt_controller_up_total":                    "",
+	"kubevirt_virt_handler_up_total":                       "",
+	"kubevirt_virt_operator_leading_total":                 "",
+	"kubevirt_virt_operator_ready_total":                   "",
+	"kubevirt_virt_operator_up_total":                      "",
+	"kubevirt_vmi_cpu_affinity":                            "",
+	"kubevirt_vmi_filesystem_capacity_bytes_total":         "",
+	"kubevirt_vmi_memory_domain_bytes_total":               "",
+	"kubevirt_vmi_memory_pgmajfault":                       "",
+	"kubevirt_vmi_memory_pgminfault":                       "",
+	"kubevirt_vmi_memory_swap_in_traffic_bytes_total":      "",
+	"kubevirt_vmi_memory_swap_out_traffic_bytes_total":     "",
+	"kubevirt_vmi_outdated_count":                          "",
+	"kubevirt_vmi_phase_count":                             "",
+	"kubevirt_vmi_storage_flush_times_ms_total":            "",
+	"kubevirt_vmi_storage_read_times_ms_total":             "",
+	"kubevirt_vmi_storage_write_times_ms_total":            "",
+	"kubevirt_vmi_vcpu_seconds":                            "",
+	"kubevirt_vmi_vcpu_wait_seconds":                       "",
+	"kubevirt_vmsnapshot_disks_restored_from_source_total": "",
+}
```

Comment:
this is like a set actually. You only want to know if a string is included in this map. So the best practice for these cases is to use empty struct values, as they consumes no memory. e.g.
```suggestion
var excludedMetrics = map[string]struct{}{
	"kubevirt_allocatable_nodes_count":                     struct{}{},
	"kubevirt_kvm_available_nodes_count":                   struct{}{},
	"kubevirt_migrate_vmi_pending_count":                   struct{}{},
	"kubevirt_migrate_vmi_running_count":                   struct{}{},
	"kubevirt_migrate_vmi_scheduling_count":                struct{}{},
	"kubevirt_virt_api_up_total":                           struct{}{},
	"kubevirt_virt_controller_ready_total":                 struct{}{},
	"kubevirt_virt_controller_up_total":                    struct{}{},
	"kubevirt_virt_handler_up_total":                       struct{}{},
	"kubevirt_virt_operator_leading_total":                 struct{}{},
	"kubevirt_virt_operator_ready_total":                   struct{}{},
	"kubevirt_virt_operator_up_total":                      struct{}{},
	"kubevirt_vmi_cpu_affinity":                            struct{}{},
	"kubevirt_vmi_filesystem_capacity_bytes_total":         struct{}{},
	"kubevirt_vmi_memory_domain_bytes_total":               struct{}{},
	"kubevirt_vmi_memory_pgmajfault":                       struct{}{},
	"kubevirt_vmi_memory_pgminfault":                       struct{}{},
	"kubevirt_vmi_memory_swap_in_traffic_bytes_total":      struct{}{},
	"kubevirt_vmi_memory_swap_out_traffic_bytes_total":     struct{}{},
	"kubevirt_vmi_outdated_count":                          struct{}{},
	"kubevirt_vmi_phase_count":                             struct{}{},
	"kubevirt_vmi_storage_flush_times_ms_total":            struct{}{},
	"kubevirt_vmi_storage_read_times_ms_total":             struct{}{},
	"kubevirt_vmi_storage_write_times_ms_total":            struct{}{},
	"kubevirt_vmi_vcpu_seconds":                            struct{}{},
	"kubevirt_vmi_vcpu_wait_seconds":                       struct{}{},
	"kubevirt_vmsnapshot_disks_restored_from_source_total": struct{}{},
}
```